pkg/apps/application-list.jsx

```diff
@@ -165,8 +195,13 @@ export const ApplicationList = ({ metainfo_db, appProgress, appProgressTitle, ac
                 </Flex>
             </PageSection>
             {comps.length == 0
-                ? <EmptyStatePanel title={ _("No applications installed or available.") } />
+                ? <EmptyStatePanel title={ _("No applications installed or available.") }
+                                   paragraph={data_missing_msg}
+                                   action={_("Install application information")} onAction={refresh} />
```

Comment:
Should this button always be here, even if `data_missing_msg` is null?

Reply:
Ouch, thanks for spotting! Fixed, and added a test (which took a bit of work to test a negative)